theano/tensor/nnet/nnet.py

```diff
@@ -93,23 +93,25 @@ def perform(self, node, input_storage, output_storage):
         if x.dtype == 'float16':
             x = x.astype('float32')
 
-        x_plus_b = x + b[None, :]
-        e_x = np.exp(x_plus_b - x_plus_b.max(axis=1)[:, None])
-        e_x *= 1.0 / e_x.sum(axis=1)[:, None]
+        # We add bias on the last dimension
+        x_plus_b = x + b
```

Comment:
Hi @bordesf ,

Here, you cannot sum x and b as they don't have the same shape. You need to make it broadcastable first. Also you need to change the C code accordingly.
A test case:
```python
a = T.tensor3('a')
b = T.vector('b')
res =  softmax_with_bias(a, b, 0)
f = theano.function([a, b], res)
a_ = np.random.rand(3, 4, 5)
b_ = np.random.rand([3, 4, 5][axis])
res_ = f(a_, b_)
```
```
ValueError: operands could not be broadcast together with shapes (3,4,5) (3,) 
Apply node that caused the error: SoftmaxWithBias(a, b, TensorConstant{0})
Toposort index: 0
Inputs types: [TensorType(float64, 3D), TensorType(float64, vector), TensorType(int8, scalar)]
Inputs shapes: [(3, 4, 5), (3,), ()]
Inputs strides: [(160, 40, 8), (8,), ()]
Inputs values: ['not shown', array([ 0.66083873,  0.16459271,  0.09435837]), array(0, dtype=int8)]
Outputs clients: [['output']]```